applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -342,22 +370,44 @@ static void syncable_cb(struct bt_bap_broadcast_sink *sink, bool encrypted)
 	strncpy(bis_encryption_key, CONFIG_BT_AUDIO_BROADCAST_ENCRYPTION_KEY, 16);
 #endif /* (CONFIG_BT_AUDIO_BROADCAST_ENCRYPTED) */
 
-	if (!playing_state) {
-		LOG_DBG("Syncable received, but either in paused_state or already in a stream");
+	if (active_stream.stream != NULL && active_stream.stream->ep != NULL) {
+		if (active_stream.stream->ep->status.state == BT_BAP_EP_STATE_STREAMING) {
+			LOG_WRN("Syncable received, but already in a stream");
+			return;
+		}
+	} else {
+		LOG_WRN("Stream or ep is NULL");
+	}
+
+	if (paused) {
+		LOG_DBG("Syncable received, but in paused state");
 		return;
 	}
 
 	LOG_INF("Syncing to broadcast stream index %d", active_stream_index);
 
+	if (bis_index_bitfields[active_stream_index] == 0) {
+		LOG_ERR("No bits set in bitfield");
+		return;
+	} else if (!IS_POWER_OF_TWO(bis_index_bitfields[active_stream_index])) {
+		/* Check that only one bit is set */
+		LOG_ERR("Application syncs to only one stream");
+		return;
+	}
+
+	struct bt_bap_stream *audio_streams_p[] = { &audio_streams[active_stream_index] };
```

Comment:
Should this struct be at the top of the function?

Reply:
Done